lib/travis/api/app.rb

```diff
@@ -81,11 +82,7 @@ def self.new(options = {})
     end
 
     def self.deploy_sha
-      @deploy_sha ||= File.exist?(deploy_sha_path) ? File.read(deploy_sha_path)[0..7] : 'deploy-sha'
-    end
-
-    def self.deploy_sha_path
-      File.expand_path('../../../../.deploy-sha', __FILE__)
+      @deploy_sha ||= ENV['HEROKU_SLUG_COMMIT'] || SecureRandom.hex(5)
```

Comment:
@drogus Why do we need a random hex when `ENV['HEROKU_SLUG_COMMIT']` is `null`? It could lead to misinformation, IMHO.